### PR TITLE
Monitor database for new measurements

### PR DESCRIPTION
**Pull request by FedericoCeratto**
Implements https://github.com/ooni/backend/issues/219

### CODE REVIEW

**Comment by FedericoCeratto** [comment]
https://github.com/ooni/pipeline/pull/205

File: af/analysis/analysis/analysis.py [hunk 1]
```diff
@@ -942,11 +945,140 @@ def generate_slow_query_summary(conf):
...
+
+    Runs in a dedicated thread and writes in its own .prom file
+
+    This is the most important function, therefore it pings the SystemD watchdog
+    """
+    log.info("Started monitor_measurement_creation thread")
+    # TODO: switch to OOID
+
+    INTERVAL = 60 * 5  # half of the watchdog interval in the analysis.service file
+    nodeexp_path = "/run/nodeexp/db_metrics.prom"
```
Comment (hellais) [inline af/analysis/analysis/analysis.py:969]: Do we want to have this hardcoded in here?

File: af/analysis/analysis/analysis.py [hunk 2]
```diff
@@ -942,11 +945,140 @@ def generate_slow_query_summary(conf):
...
+    if no_systemd == False:
+        watchdog.notify("READY=1")
+
+    cycle_seconds = 0
+
+    while True:
+        try:
+            log.info("Gathering fastpath count")
+            conn, dbengine = setup_database_connections(conf.standby)
+            delta = timedelta(minutes=5)
```
Comment (hellais) [inline af/analysis/analysis/analysis.py:1039]: Maybe `5` should be a variable which is in sync with `INTERVAL = 60 * 5`?

File: af/analysis/analysis/analysis.py [hunk 3]
```diff
@@ -942,11 +945,140 @@ def generate_slow_query_summary(conf):
...
 
+@metrics.timer("monitor_measurement_creation")
+def monitor_measurement_creation(conf):
+    """Monitor measurements created by fastpath and traditional pipeline
+    to detect and alert on inconsistencies
+
+    Runs in a dedicated thread and writes in its own .prom file
+
+    This is the most important function, therefore it pings the SystemD watchdog
+    """
```
Comment (hellais) [inline af/analysis/analysis/analysis.py:964]: It's perhaps useful to add a comment on the logic of the queries that follow, as in that we are sampling every interval and comparing the number of measurements in the sample window period in the fastpath and the batch pipeline.

**Review by hellais — COMMENTED** [review]
Left some review comments.

File: af/analysis/analysis/analysis.py (hunk 3 above)
Comment (FedericoCeratto) [reply af/analysis/analysis/analysis.py:964]: Added

File: af/analysis/analysis/analysis.py (hunk 1 above)
Comment (FedericoCeratto) [reply af/analysis/analysis/analysis.py:969]: Yes, it's close to where it's being used. monitor_measurement_creation() is meant to be self-contained

File: af/analysis/analysis/analysis.py (hunk 2 above)
Comment (FedericoCeratto) [reply af/analysis/analysis/analysis.py:1039]: No, this is independent from the run interval, e.g. we could run it every minute.

**Review by hellais — APPROVED** [review]
LGTM 🐳 !

**Pull request merged**